build.gradle

```diff
@@ -50,21 +50,37 @@ subprojects {
       javadoc = true
       tests = false
     }
+    nexus {
+      def gitProvider = new nebula.plugin.info.scm.GitScmProvider()
+      def shortHash = gitProvider.calculateChange(new File('.'))
+      def isRelease = System.getProperty('build.snapshot', 'true') == 'false'
+      if (isRelease) {
+        repositoryUrl = System.getProperty('build.repository', "file://${System.getenv('HOME')}/elasticsearch-releases/${version}-${shortHash}/")
+      }
+    }
     // we have our own username/password prompts so that they only happen once
-    // TODO: add gpg signing prompts
+    // TODO: add gpg signing prompts, which is tricky, as the buildDeb/buildRpm tasks are executed before this code block
```

Comment:
Yes we can probably do some magic...it looks like the signing support was added ospackage in a non-standard way. Throughout gradle APIs, you see settings like this as Object so that they can be lazily evaluated. Here that would have allowed setting a placeholder object, which the impl of could have had the value filled in for.  Perhaps we should think about a PR back to ospackage to do that (change the types for the [signing key id and passphrase](https://github.com/nebula-plugins/gradle-ospackage-plugin/commit/0d484c764ce72dcb33f695e5de4f1f9ccb663bc6#diff-4c98103916724963b2b69c0918249f1eR36) to Object).